packages/core/admin/admin/src/content-manager/pages/ListView/components/Filters.tsx

```diff
@@ -0,0 +1,291 @@
+import * as React from 'react';
+
+import { Button } from '@strapi/design-system';
+import {
+  FilterData,
+  FilterListURLQuery,
+  FilterPopoverURLQuery,
+  findMatchingPermissions,
+  useCollator,
+  useQueryParams,
+  useRBACProvider,
+  useTracking,
+} from '@strapi/helper-plugin';
+import { Filter as FilterIcon } from '@strapi/icons';
+import { useIntl } from 'react-intl';
+
+import { useEnterprise } from '../../../../hooks/useEnterprise';
+import { useAdminUsers } from '../../../../services/users';
+import { CREATOR_FIELDS } from '../../../constants/attributes';
+import { Schema } from '../../../hooks/useDocument';
+import { useGetContentTypeConfigurationQuery } from '../../../services/contentTypes';
+import { getDisplayName } from '../../../utils/users';
+
+import { AdminUsersFilter } from './AdminUsersFilter';
+
+const REVIEW_WORKFLOW_FILTER_CE: FilterData[] = [];
+
+/**
+ * If new attributes are added, this list needs to be updated.
+ */
+const NOT_ALLOWED_FILTERS = [
+  'json',
+  'component',
+  'media',
+  'richtext',
+  'dynamiczone',
+  'password',
+  'blocks',
+];
+const DEFAULT_ALLOWED_FILTERS = ['createdAt', 'updatedAt'];
+const USER_FILTER_ATTRIBUTES = [...CREATOR_FIELDS, 'strapi_assignee'];
+
+interface FiltersProps {
+  disabled?: boolean;
+  schema: Schema;
+}
+
+const Filters = ({ disabled, schema }: FiltersProps) => {
+  const { attributes, uid: model, options } = schema;
+  const [isVisible, setIsVisible] = React.useState(false);
+  const { formatMessage, locale } = useIntl();
+  const buttonRef = React.useRef<HTMLButtonElement>(null!);
+  const { trackUsage } = useTracking();
+  const { allPermissions } = useRBACProvider();
+  const [{ query }] = useQueryParams<{
+    filters?: {
+      $and: Array<{
+        [key: string]: {
+          id?: {
+            $eq?: string;
+            $ne?: string;
+          };
+        };
+      }>;
+    };
+  }>();
+
+  const canReadAdminUsers = React.useMemo(
+    () =>
+      findMatchingPermissions(allPermissions, [
+        {
+          action: 'admin::users.read',
+          subject: null,
+        },
+      ]).length > 0,
+    [allPermissions]
+  );
+
+  const selectedUserIds =
+    query?.filters?.$and?.reduce<string[]>((acc, filter) => {
+      const [key, value] = Object.entries(filter)[0];
+      const id = value.id?.$eq || value.id?.$ne;
+      // TODO: strapi_assignee should not be in here and rather defined
+      // in the ee directory.
+      if (id && USER_FILTER_ATTRIBUTES.includes(key) && !acc.includes(id)) {
```

Comment:
I think next comment would make more sense next to line 41 where strapi_assignee is mentioned